### PR TITLE
Adopt `dni_clear` variable name for clearsky DNI

### PR DESCRIPTION
**Pull request by RDaxini**
- [X] Related: #2272
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests ~added~ modified
 - [ ] ~Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [X] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

`dni_clear` is already in the variables and symbols list
Also related: #1253, #2306

### CODE REVIEW

**Comment by AdamRJensen** [comment]
In both this PR and in #2273, you should use the decorate added in #2237.

The reason is that simply changing the parameter name would be a breaking change.

**Comment by RDaxini** [comment]
> In both this PR and in #2273, you should use the decorate added in #2237.
> 
> The reason is that simply changing the parameter name would be a breaking change.

ah I asked about this in the linked issue as I thought it might be, thanks @AdamRJensen

File: pvlib/tests/test_irradiance.py [hunk 1]
```diff
@@ -15,7 +15,8 @@
     assert_frame_equal,
     assert_series_equal,
     requires_ephem,
-    requires_numba
+    requires_numba,
+    fail_on_pvlib_version
```
Comment (AdamRJensen) [inline pvlib/tests/test_irradiance.py:19]:
```suggestion
    fail_on_pvlib_version,
```
Adding a comma avoids the line showing up in the diff when adding a new entry to the list (see for example how the requires_numba is showing up).

File: pvlib/irradiance.py [hunk 2]
```diff
@@ -2151,7 +2151,12 @@ def _dirint_bins(times, kt_prime, zenith, w, delta_kt_prime):
     return kt_prime_bin, zenith_bin, w_bin, delta_kt_prime_bin
 
 
-def dirindex(ghi, ghi_clearsky, dni_clearsky, zenith, times, pressure=101325.,
+@renamed_kwarg_warning(
+    since='0.11.2',
+    old_param_name='dni_clearsky',
+    new_param_name='dni_clear',
+    removal="0.12.0")
```
Comment (kandersolar) [inline pvlib/irradiance.py:2158]: I'm hesitant to say we'll remove the deprecations this soon.  0.12.0 could potentially be in the next couple releases, which means the deprecation period would only be 3 or 6 months.  I don't think pvlib has a firm convention on this, but I remember saying "at least 12 months" previously.  

Input from others would be helpful here.

File: pvlib/irradiance.py [hunk 3]
```diff
@@ -3674,14 +3684,19 @@ def dni(ghi, dhi, zenith, clearsky_dni=None, clearsky_tolerance=1.1,
...
             (zenith < zenith_threshold_for_zero_dni) &
             (dni > max_dni)] = max_dni
     return dni
 
 
+@renamed_kwarg_warning(
+    since='0.11.2',
+    old_param_name='clearsky_dni',
+    new_param_name='dni_clear',
+    removal="0.12.0")
```
Comment (kandersolar) [inline pvlib/irradiance.py:3699]: No need for this change, right?  We're not renaming any parameters for `complete_irradiance`.

File: pvlib/tests/test_irradiance.py [hunk 4]
```diff
@@ -1129,10 +1130,30 @@ def test_dni():
...
+    dhi = pd.Series([100, 90, 50, 50, 50])
+    zenith = pd.Series([80, 100, 85, 70, 85])
+    dni_clear = pd.Series([50, 50, 200, 50, 300])
+
+    with pytest.warns(pvlibDeprecationWarning, match='dni_clear'):
+        dni = irradiance.dni(ghi, dhi, zenith,
+                             clearsky_dni=dni_clear, clearsky_tolerance=2)
     assert_series_equal(dni,
                         pd.Series([float('nan'), float('nan'), 400,
                                    146.190220008, 573.685662283]))
```
Comment (kandersolar) [inline pvlib/tests/test_irradiance.py:1159]: No need to check the return values for these deprecation tests IMHO.  The point is to verify that the warning is raised.

File: pvlib/tests/test_irradiance.py [hunk 5]
```diff
@@ -1129,10 +1130,30 @@ def test_dni():
...
+                                   146.190220008, 573.685662283]))
+
+    dni = irradiance.dni(ghi, dhi, zenith)
+    assert_series_equal(dni,
+                        pd.Series([float('nan'), float('nan'), 573.685662283,
+                                   146.190220008, 573.685662283]))
+
+
+@fail_on_pvlib_version("0.12")
+def test_dni_dni_clearsky_deprecation():
```
Comment (kandersolar) [inline pvlib/tests/test_irradiance.py:1148]: We should add a similar `with pytest.warns` check for the `dirindex` function.  Can be in this same test function or a new one, up to you.

**Review by kandersolar — COMMENTED** [review]
These PRs will certainly need a whatsnew entry.

Has anyone audited `iotools` to see if we can standardize any clearsky DNI names there too?

File: docs/sphinx/source/whatsnew/v0.11.2.rst [hunk 6]
```diff
@@ -6,6 +6,8 @@ v0.11.2 (Anticipated December, 2024)
 
 Deprecations
 ~~~~~~~~~~~~
+* Deprecated terms `dni_clearsky` and `clearsky_dni`, replaced with `dni_clear`
```
Comment (echedey-ls) [inline docs/sphinx/source/whatsnew/v0.11.2.rst:9]:
```suggestion
* Deprecated terms ``dni_clearsky`` and ``clearsky_dni``, replaced with ``dni_clear``.
  Affected functions are :py:func:`~pvlib.irradiance.dirindex` and :py:func:`~pvlib.irradiance.dni`.
```

File: pvlib/irradiance.py (hunk 2 above)
Comment (echedey-ls) [reply pvlib/irradiance.py:2158]: I think there is no harm in supporting old names for 12 months. +1 for `v0.13.0`.

**Review by echedey-ls — COMMENTED** [review]
Mostly rST comments. PR looks good 🚀 

Shall any `.. deprecated` or `.. versionchanged` directives be used? I'm +1 for the latter, I have the mild impression the former is for complete removals.

Docs at: https://pvlib-python--2274.org.readthedocs.build/en/2274/
- https://pvlib-python--2274.org.readthedocs.build/en/2274/whatsnew.html
- https://pvlib-python--2274.org.readthedocs.build/en/2274/reference/generated/pvlib.irradiance.dni.html#pvlib-irradiance-dni

**Comment by RDaxini** [comment]
> Has anyone audited `iotools` to see if we can standardize any clearsky DNI names there too?

Variable maps exist.
`psm3.py`, `solargis.py`, `sodapro.py`, `solcast.py`, and `solaranywhere.py` map to `*_clear`, e.g:
https://github.com/pvlib/pvlib-python/blob/6af80da35a7c96059c534ee38be9123bcfc7f50f/pvlib/iotools/psm3.py#L28-L30
https://github.com/pvlib/pvlib-python/blob/6af80da35a7c96059c534ee38be9123bcfc7f50f/pvlib/iotools/solargis.py#L29-L37

I'm not 100% familiar with how these maps work but I guess instances of other variable names that are part of the map, e.g.:
https://github.com/pvlib/pvlib-python/blob/6af80da35a7c96059c534ee38be9123bcfc7f50f/pvlib/iotools/solcast.py#L355-L363
do not matter because they will be mapped to `*_clear`(?)

I think the variable name on the left is external and right is pvlib internal, but then it looks like `*_clear` was already there before it was used in these functions in pvlib...? I might have misunderstood something here.

Can someone advise if additional revisions are needed?

**Comment by kandersolar** [comment]
Thanks for checking.  I think no action is needed in iotools then. 

> Shall any .. deprecated or .. versionchanged directives be used? I'm +1 for the latter, I have the mild impression the former is for complete removals.

Yes, I think so. Right now the docstring doesn't have any indication that the name changed.  I like how @echedey-ls did it here: https://github.com/echedey-ls/pvlib-python/blob/1eb44a5d5788b398ae3333ae5f96a04df6567dea/pvlib/solarposition.py#L1351-L1356

**Review by kandersolar — APPROVED** [review]
LGTM.  Any comments from others (@echedey-ls?)

**Review by echedey-ls — APPROVED** [review]
LGTM too!

**Comment by echedey-ls** [comment]
After some thought, I think stating the version when the legacy param names will be removed would be useful, but neither essential nor a blocker for merging this PR. We can do that in the future or after taking care of #2325.

**Review by AdamRJensen — APPROVED** [review]

**Pull request merged**